examples/decoding/plot_ssd_spatial_filters.py

```diff
@@ -28,144 +25,378 @@
 import mne
 from mne import Epochs
 from mne.datasets.fieldtrip_cmc import data_path
-from mne.utils import _time_mask
 from mne.channels import read_layout
 from mne.decoding import TransformerMixin, BaseEstimator
 
+from mne.io.base import BaseRaw
+from mne.epochs import BaseEpochs
+from mne.utils import _time_mask
+from mne.cov import (_regularized_covariance, compute_raw_covariance)
+from mne.filter import filter_data
+from mne.time_frequency import psd_array_welch
+import locale
+locale.setlocale(locale.LC_ALL, "en_US.UTF-8")
+
 
 def freq_mask(freqs, fmin, fmax):
     """convenience function to select frequencies"""
     return _time_mask(freqs, fmin, fmax)
 
+
 # Define parameters
 fname = data_path() + '/SubjectCMC.ds'
 raw = mne.io.read_raw_ctf(fname)
 raw.crop(50., 250.).load_data()  # crop for memory purposes
+picks = mne.pick_types(raw.info, meg=True, eeg=False, ref_meg=False)
+sf = raw.info['sfreq']
 
 freqs_sig = 9, 12
 freqs_noise = 8, 13
 
 
 class SSD(BaseEstimator, TransformerMixin):
-    """Implement Spatio-Spectral Decomposision
+    """
+    This is a Python Implementation of Spatio Spectral Decomposition (SSD)
+    method [1],[2] for both raw and epoched data. This source code is  based on
+    the matlab implementation available at
+    https://github.com/svendaehne/matlab_SPoC/tree/master/SSD and the
+    PR MNE-based implementation of Denis A. Engemann <denis.engemann@gmail.com>
```

Comment:
These authorship details are not necessary. You do not find this in other functions of the MNE source code.
For what I proposed at least, it is also not true that the source code was based on any other code. I implemented the initial code based on the 2011 paper. Let me know if the reference to the Matlab code is justified for any line that you have proposed.